### ISSUE TITLE
Use the type manager dropdown to change the type in the scene. 

### ISSUE THREAD

**Issue by theoryshaw** [IfcOpenShell/IfcOpenShell#3707]
Intuitively it feels like doing something like the following video, changes the type in the scene. 

That is, when a type is selected, you can use the type manager dropdown to change the type in the scene. 

https://github.com/IfcOpenShell/IfcOpenShell/assets/507113/a45808d1-4dec-467d-9b6d-f660559048f0

**Comment by Gorgious56** [comment]
I am personally not sure about this, since the goal of the type manager as far as I understand is to select and manage types for instanciation, not for managing existing individual instances. I would be fine with a button placed in the manager header to update the selected instance(s) to the type that is selected in the manager though.

I may have expectations based on how it works now though so maybe I'm biased towards keeping it that way. :)

**Comment by myoualid** [comment]
I'm also biased and got used to how things work. But maybe there's a middle ground:

- The dropdown in the Type Manager (from the top bar) shows the types to be added to the scene.
- BUT, In the Active Tool, the dropdown shows the type of the highlighted object + when you change, it changes the type of all selected objects. 

Or vice versa

**Comment by theoryshaw** [comment]
> the scene.
BUT, In the Active Tool, the dropdown shows the type of the highlighted object + when you change, it changes the type of all selected objects

yes, was thinking something like this.

**Comment by trhyder** [comment]
Thanks for the ping.

It's good to hear this is getting some love. I had a look and here are my suggestions.

<img width="570" alt="Screenshot 2024-09-23 at 11 37 09 AM" src="https://github.com/user-attachments/assets/6a3f24ef-283b-4206-9ec8-b0a5fb3215e0">

<img width="313" alt="Screenshot 2024-09-23 at 11 37 34 AM" src="https://github.com/user-attachments/assets/614c7ab5-f55f-4b5e-ab06-65c80b54e780">

Additionally, it would be great to access other fields like Profiles and Material Layer Sets from the Types Manager.

**Comment by Moult** [comment]
The two buttons at the bottom "Cancel" and "Add Type" (the text defaults to "OK", clicking it will execute "something") are a sort of necessary evil and I'd love ideas on how to improve it.

Basically Blender by default has a few different ways to pop things up. A regular popup has no "Cancel/OK" button at the bottom, but if you move your mouse away, it disappears and auto executes. This is really disturbing for both new and old users. A dialog popup doesn't have this "mouse movement" problem, but it does come with a "Cancel/OK" combo which we can't remove (but we can change the "OK" text to something else and control what it executes). So ideally we come up with an idea for those two buttons to make them meaningful.

In short as you probably already know Blender's interface doesn't encourage popup dialogs, so in the few areas where we use them, we need to use them carefully.

**Comment by trhyder** [comment]
Hey @Moult my PR https://github.com/IfcOpenShell/IfcOpenShell/pull/5500 got behind the work you had done. I would like to contribute more in this area. Below is a screen capture of what I had done, if you like any of it let me know and I'll create a new PR.

![ScreenRecording2024-10-02at10 13 35PM-ezgif com-video-to-gif-converter (1)](https://github.com/user-attachments/assets/285e1df5-12a6-4d5d-bc5c-d38f337801d7)

**Comment by Moult** [comment]
@trhyder would it be easier if we organised a screenshare to go through the merge conflicts and try to set some boundaries to prevent future merge conflicts?

**Comment by trhyder** [comment]
@moult, do you see any problems addressing @theoryshaw 's initial idea - to use the Type Manager to change the ElementType of a selected object?

**Comment by Moult** [comment]
I think that's totally a fine idea, especially now that we have distinct add / edit UIs.
